src/main/java/org/wso2/carbon/connector/utils/SnowflakeUtils.java

```diff
@@ -0,0 +1,194 @@
+import org.wso2.carbon.connector.exception.InvalidConfigurationException;
+import org.wso2.carbon.connector.pojo.SnowflakesOperationResult;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Util methods related to snowflake connector operations
+ */
+public class SnowflakeUtils {
+
+    private static final Log log = LogFactory.getLog(SnowflakeUtils.class);
+
+    /**
+     * Get the connection name from the message context
+     *
+     * @param messageContext
+     * @return connection name
+     * @throws InvalidConfigurationException
+     */
+    public static String getConnectionName(MessageContext messageContext) throws InvalidConfigurationException {
+        String connectionName = (String) messageContext.getProperty(Constants.CONNECTION_NAME);
+        if (StringUtils.isEmpty(connectionName)) {
+            throw new InvalidConfigurationException("Connection name is not set.");
+        }
+        return connectionName;
+    }
+
+    /**
+     * Set the result as the payload in the message context
+     *
+     * @param msgContext
+     * @param result
+     */
+    public static void setResultAsPayload(MessageContext msgContext, SnowflakesOperationResult result) {
+        JsonObject jsonObject = new JsonObject();
+        jsonObject.addProperty("operation", result.getOperation());
+        jsonObject.addProperty("isSuccessful", result.isSuccessful());
+
+        if (result.getMessage() != null && !result.getMessage().isEmpty()) {
+            jsonObject.addProperty("message", result.getMessage());
+        }
+
+        if (result.getError() != null) {
+            setErrorPropertiesToMessageContext(msgContext, result.getError());
+            jsonObject.addProperty("error", result.getError().getErrorCode());
+            jsonObject.addProperty("code", result.getError().getErrorCode());
+            jsonObject.addProperty("errorDetail", result.getError().getErrorDetail());
+        }
+
+        org.apache.axis2.context.MessageContext axisMsgCtx =
+                ((Axis2MessageContext) msgContext).getAxis2MessageContext();
+
+        JsonUtil.removeJsonPayload(axisMsgCtx);
+        axisMsgCtx.removeProperty("NO_ENTITY_BODY");
+        try {
+            JsonUtil.getNewJsonPayload(axisMsgCtx, jsonObject.toString(), true, true);
+        } catch (AxisFault axisFault) {
+            log.error("Error occurred while populating the payload in the message context.", axisFault);
+        }
+    }
+
+    /**
+     * Set the result as the payload in the message context
+     *
+     * @param msgContext message context
+     * @param result the result array to be set as the payload
+     */
+    public static void setResultAsPayload(MessageContext msgContext, JsonArray result) {
+        org.apache.axis2.context.MessageContext axisMsgCtx =
+                ((Axis2MessageContext) msgContext).getAxis2MessageContext();
+        JsonUtil.removeJsonPayload(axisMsgCtx);
+        axisMsgCtx.removeProperty("NO_ENTITY_BODY");
```

Comment:
are we removing NO_ENTITY_BODY in other connecters also ? or let the user remove it ?

Reply:
We have done the same in file connector too.
https://github.com/wso2-extensions/esb-connector-file/blob/ae1912693b3e546b598de8e8e162b66bc22e8df7/src/main/java/org/wso2/carbon/connector/utils/Utils.java#L266C1-L267